lib/eventboss/sns_client.rb

```diff
@@ -10,10 +13,33 @@ def publish(payload)
       backend.publish(payload)
     end
 
+    def create_topic(topic_name)
+      backend.create_topic(name: topic_name)
+    end
+
+    def create_subscription(topic_arn, queue_arn)
+      subscription = backend.subscribe(
+        topic_arn: topic_arn,
+        endpoint: queue_arn,
+        protocol: 'sqs'
+      )
+      set_raw_message_delivery(subscription)
+    end
+
     private
 
     attr_reader :configuration
 
+    def set_raw_message_delivery(subscription)
+      [false, true].each do |bool|
+        backend.set_subscription_attributes(
+          subscription_arn: subscription.subscription_arn,
+          attribute_name: 'RawMessageDelivery',
+          attribute_value: bool.to_s
+        )
+      end
```

Comment:
Could you elaborate/document why?

Reply:
Actually setting it to false first is not needed any more.